frontend/src/pages/projects/types.ts

```diff
@@ -9,14 +9,16 @@ import {
   Volume,
   VolumeMount,
 } from '~/types';
-import { ValueOf } from '~/typeHelpers';
 import { AWSSecretKind } from '~/k8sTypes';
 import { AcceleratorProfileState } from '~/utilities/useAcceleratorProfileState';
 import { AcceleratorProfileSelectFieldState } from '~/pages/notebookController/screens/server/AcceleratorProfileSelectField';
 import { K8sNameDescriptionFieldData } from '~/concepts/k8s/K8sNameDescriptionField/types';
 import { AwsKeys } from './dataConnections/const';
 
-export type UpdateObjectAtPropAndValue<T> = (propKey: keyof T, propValue: ValueOf<T>) => void;
+export type UpdateObjectAtPropAndValue<T> = <K extends keyof T>(
+  propKey: K,
+  propValue: Partial<T>[K],
```

Comment:
Won't this break the sanctity of our type behind the object store?

 I can send `undefined` to a value required by the type.

Reply:
The current types are already broken in that I can supply a value from _any_ prop for _any_ key.
This isn't a perfect solution but a step that's better IMO. Enforces correct type but is allowing `undefined`.

I suppose we can always keep looking for the perfect solution...

Reply:
Went with another approach that required altering several components which utilize this type. I was not able to find a solution whereby only the `UpdateObjectAtPropAndValue` could be updated and solve all use cases.